src/integrationtests/java/com/aws/greengrass/integrationtests/extensions/BridgeIntegrationTestExtension.java

```diff
@@ -210,6 +232,42 @@ private void pointBridgeToV5Broker() throws ServiceLoadException, InterruptedExc
         assertTrue(bridgeRunning.await(30L, TimeUnit.SECONDS));
     }
 
+    @SuppressWarnings("PMD.TooFewBranchesForASwitchStatement")
+    private void waitForClientsToConnect() throws InterruptedException {
+        CountDownLatch localClientConnected = new CountDownLatch(1);
+        AtomicReference<Exception> error = new AtomicReference<>();
+        context.getFromContext(ExecutorService.class).submit(
+                () -> {
+                    // TODO can improve on this by having clients reported their connected status within bridge itself
+                    switch (context.getConfig().getMqttVersion()) {
+                        case MQTT5:
+                            error.set(new UnsupportedOperationException("TODO add support"));
+                            break;
+                        case MQTT3:
+                            while (!context.getLocalV3Client().getMqttClientInternal().isConnected()) {
+                                try {
+                                    Thread.sleep(100L);
+                                } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)

Reply:
resolving, test will fail on interrupted